### ISSUE TITLE
Upgrade go-netbox for v3.0 support

### ISSUE THREAD

**Issue by passcod** [smutel/terraform-provider-netbox#66]
## Summary

Now that go-netbox has a Netbox 3.0 release: https://github.com/netbox-community/go-netbox/releases/tag/netbox_v3.0 it would be great to update this provider!

**Comment by BonySmoke** [comment]
Also, @smutel, I would be glad to help with the upgrade if I can.
Although it won't be possible to use secrets anymore, the patch contains a fix for the bug with interfaces ([Issue 100 in project go-netbox](https://github.com/netbox-community/go-netbox/issues/100))

**Comment by smutel** [comment]
For now I don't want to focus my effort on Netbox 3.0 because it's a very young major release. I want to release my branch `Netbox2.11`. I refactored the custom fields part.

I decided also to create my own [go-netbox](https://github.com/smutel/go-netbox) project to not be dependent from other project. I created my project Sept 2 and they created a new release Sept 6 ... We will see in the future if I continue to use my own project or not. I integrated from the beginning the generation of library by go-swagger in the CI and the patchs applied to the swagger are in separated files. I think it's a better approach to speed up the release.

@BonySmoke Thank you for your (previous) help. If you or someone want to help at this point, I think the best will be to add some automated tests to this project.
